src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,59 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+#ifdef ESCARGOT_ENABLE_ES2015
+static Value builtinRegExpMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(thisObject, RegExp, match);
+
+    String* str = argv[0].toString(state);
+    ASSERT(str != nullptr);
+    RegExpObject* rx;
+
+    if (thisObject->isRegExpObject(state) == true) {
+        rx = thisObject->asRegExpObject(state);
+    } else {
+        rx = new RegExpObject(state, Value(thisObject).toString(state), String::emptyString);
+    }
+    //21.2.5.6.8
+    bool global = rx->option() & RegExpObject::Option::Global;
```

Comment:
This is not equivalent with: `Let global be ToBoolean(Get(rx, "global")).`, since the named regexp flags properties can be overwritten during the execution, which does not have an affect on the internal flags.